packages/grid/x-data-grid/src/tests/columnHeaders.DataGrid.test.tsx

```diff
@@ -49,4 +49,44 @@ describe('<DataGrid /> - Column Headers', () => {
       expect(getColumnHeaderCell(0)).to.have.class('foobar');
     });
   });
+
+  describe('header menu', () => {
+    it('should allow to hide column', () => {
+      const { getByRole, getAllByLabelText } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            columns={[{ field: 'id' }, { field: 'brand', headerClassName: 'foobar' }]}
+          />
+        </div>,
+      );
+
+      expect(getColumnHeadersTextContent()).to.deep.equal(['id', 'brand']);
+
+      fireEvent.click(getAllByLabelText('Menu')[0]);
+      fireEvent.click(getByRole('menuitem', { name: 'Hide' }));
+
+      clock.runToLast();
+      expect(getColumnHeadersTextContent()).to.deep.equal(['brand']);
+    });
+
+    it('should not allow to hide the only visible column', () => {
+      const { getByRole, getAllByLabelText } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            columns={[{ field: 'id' }, { field: 'brand', headerClassName: 'foobar', hide: true }]}
+          />
+        </div>,
+      );
+
+      expect(getColumnHeadersTextContent()).to.deep.equal(['id']);
+
+      fireEvent.click(getAllByLabelText('Menu')[0]);
+      fireEvent.click(getByRole('menuitem', { name: 'Hide' }));
```

Comment:
I'm struggling to make this test pass.

When clicking manually in the browser, disabled `Hide` button doesn't trigger `click` event, but in this test it does.

`Hide` button is a `ButtonBase` component rendered as a `li` element.
Since it's not a `button`, it doesn't have a `disabled` HTML attribute when disabled. Instead, it has `aria-disabled` + `pointer-events: none`.

I've noticed that imperative `.click()` called on disabled `li` `ButtonBase` does trigger `click` as well.
I believe same happens in tests.

Here's a minimal reproduction: https://codesandbox.io/s/disabled-buttonbase-click-c91hx?file=/src/App.js
Steps to reproduce:
1. Click `Trigger .click() on ButtonBase imperatively`
1. Nothing happens
1. Click `Change ButtonBase component to "li"`
1. onClick is called, even though `ButtonBase` is `disabled`

I'm not sure if we should consider this as a bug.

Any ideas how to make the test pass?
I've tried to use `userEvent.touch` instead of `fireEvent.click`, but it doesn't feel right.

Reply:
@alexfauquette Thanks for looking into it!

> the easiest way is probably to not test by click but only check that the button has aria-disabled

The downside I see here is that `aria-disabled` is an implementation detail, so if we change from `li` to `button` the test will fail.
Also, in the test we trust that if `aria-disabled` is present, onClick won't be triggered.
It would do the job for now though, so I think it's fine. Will add a comment explaining the decision.

Reply:
Adding a comment to explain the decision is a good idea 👍

Reply:
Do you know if the test only fails in JSDOM? It could be some limitation of JSDOM. Tests running with `yarn test:karma` use a real browser so it should behave in the same way as when manually clicking. Another thing you could do is to check if the button is disabled inside the `onClick` handler.

Reply:
> Do you know if the test only fails in JSDOM?

It fails in `test:karma`

> Another thing you could do is to check if the button is disabled inside the onClick handler

Yeah, that would allow to keep `fireEvent.click` in the test